test/eventlistener_test.go

```diff
@@ -348,3 +369,40 @@ func TestEventListenerCreate(t *testing.T) {
 	}
 	t.Log("EventListener's Service was deleted")
 }
+
+// The structure of this field corresponds to values for the `license` key in
+// testdata/pr.json, and can be used to unmarshal the dat.
+type license struct {
+	key     string
+	name    string
+	spdx_id string
+	url     string
+	node_id string
+}
+
+// compareParamsWithLicenseJSON will compare the passed in ResourceParams by further checking
+// when the values aren't equal if they can be unmarshalled into the license object and if they are
+// then equal. This is because the order of values in a dictionary is not deterministic and dictionary
+// values passed through an event listener may change order.
+func compareParamsWithLicenseJSON(x, y v1alpha1.ResourceParam) bool {
```

Comment:
Is this different from unmarshalling and then using cmp.Diff?

Reply:
Okay now im unmarshalling + then using cmp.Diff! Most of the function remains the same but now I don't have to manually check all the fields.

I feel like I'm slowly starting to understand how all the unmarshalling and marshalling works - cant wait to stop working on it and forget in 6 months X'D